apps/dashboard/src/components/primitives/segmented-control.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { Slottable } from '@radix-ui/react-slot';
+import * as TabsPrimitive from '@radix-ui/react-tabs';
+import mergeRefs from 'merge-refs';
+
+import { useTabObserver } from '@/hooks/use-tab-observer';
+import { cn } from '../../utils/ui';
+
+const SegmentedControlRoot = TabsPrimitive.Root;
+SegmentedControlRoot.displayName = 'SegmentedControlRoot';
+
+const SegmentedControlList = React.forwardRef<
+  React.ComponentRef<typeof TabsPrimitive.List>,
+  React.ComponentPropsWithoutRef<typeof TabsPrimitive.List> & {
+    floatingBgClassName?: string;
+  }
+>(({ children, className, floatingBgClassName, ...rest }, forwardedRef) => {
+  const [lineStyle, setLineStyle] = React.useState({ width: 0, left: 0 });
+
+  const { mounted, listRef } = useTabObserver({
+    onActiveTabChange: (_, activeTab) => {
+      const { offsetWidth: width, offsetLeft: left } = activeTab;
+      setLineStyle({ width, left });
+    },
+  });
```

Comment:
This new component is later needed by the create and update sidebars, I have extracted to a reusable SegmentedControl component, which is also used by the tabs component. For instances where the segmented control is used as a form element and not tabbed functionality

https://github.com/user-attachments/assets/533ae16c-0a0a-441b-9cf5-197e38dbc502